frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -153,7 +161,10 @@ export const TextField = ({
       onBlur={onChange}
       error={error}
       helperText={helpText}
-      InputProps={{ readOnly }}
+      InputProps={{
+        readOnly,
+        endAdornment: endAdornment && <IconButton type="submit">{endAdornment}</IconButton>,
```

Comment:
at first glance i wonder if we should take IconButton instead of wrapping it ourselves for further customization, this seems fine though.

Reply:
I had that thought too but I can't imagine a situation where we wouldn't want an icon button. Also, the styling applied is overriding some of the icon buttons styles which by default provide a border radius of 50% and have padding